src/GitVersionCore/VersionConverters/WixUpdater/WixVersionFileUpdater.cs

```diff
@@ -35,6 +35,7 @@ public void Execute(VersionVariables variables, WixVersionContext context)
             var root = doc.DocumentElement;
             doc.InsertBefore(xmlDecl, root);
 
+            fileSystem.Delete(wixVersionFile);
```

Comment:
Should this always be deleted, unconditionally?

Reply:
I would say yes, as the entire file is fully re-created (no update). This way it is ensured that the written file does not contain garbage after the expected content.
Maybe one could check whether the file exists, which is in my opinion useless as Delete() does not throw when the file does not exist.